central/graphql/resolvers/image_components_test.go

```diff
@@ -0,0 +1,434 @@
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	imageCompTests := []struct {
+		name        string
+		id          string
+		expectedIDs []string
+	}{
+		{
+			"comp1",
+			"comp1#0.9#",
+			[]string{"sha1", "sha2"},
+		},
+		{
+			"comp2",
+			"comp2#1.1#",
+			[]string{"sha1"},
+		},
+		{
+			"comp3",
+			"comp3#1.0#",
+			[]string{"sha1", "sha2"},
+		},
+		{
+			"comp4",
+			"comp4#1.0#",
+			[]string{"sha2"},
+		},
+	}
+
+	for _, test := range imageCompTests {
+		s.T().Run(test.name, func(t *testing.T) {
+			comp := s.getImageComponentResolver(ctx, test.id)
+			expectedCount := int32(len(test.expectedIDs))
+
+			images, err := comp.Images(ctx, PaginatedQuery{})
+			s.NoError(err)
+			s.Equal(expectedCount, int32(len(images)))
+			idList := getIDList(ctx, images)
+			s.ElementsMatch(test.expectedIDs, idList)
+
+			count, err := comp.ImageCount(ctx, RawQuery{})
+			s.NoError(err)
+			s.Equal(expectedCount, count)
+		})
+	}
+}
+
+func (s *GraphQLImageComponentTestSuite) TestImageComponentImageVulnerabilities() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	type counterValues struct {
+		fixable   int32
+		critical  int32
+		important int32
+		moderate  int32
+		low       int32
+	}
+
+	imageCompTests := []struct {
+		name                  string
+		id                    string
+		expectedIDs           []string
+		expectedCounterValues counterValues
+	}{
+		{
+			"comp1",
+			"comp1#0.9#",
+			[]string{"cve-2018-1#"},
+			counterValues{
+				1, 1, 0, 0, 0,
+			},
+		},
+		{
+			"comp2",
+			"comp2#1.1#",
+			[]string{"cve-2018-1#"},
+			counterValues{
+				1, 1, 0, 0, 0,
+			},
+		},
+		{
+			"comp3",
+			"comp3#1.0#",
+			[]string{"cve-2019-1#", "cve-2019-2#"},
+			counterValues{
+				0, 0, 0, 1, 1,
+			},
+		},
+		{
+			"comp4",
+			"comp4#1.0#",
+			[]string{"cve-2017-1#", "cve-2017-2#"},
+			counterValues{
+				0, 0, 2, 0, 0,
+			},
+		},
+	}
+
+	for _, test := range imageCompTests {
+		s.T().Run(test.name, func(t *testing.T) {
+			comp := s.getImageComponentResolver(ctx, test.id)
+			expectedCount := int32(len(test.expectedIDs))
+
+			vulns, err := comp.ImageVulnerabilities(ctx, PaginatedQuery{})
+			s.NoError(err)
+			s.Equal(expectedCount, int32(len(vulns)))
+			idList := getIDList(ctx, vulns)
+			s.ElementsMatch(test.expectedIDs, idList)
+
+			count, err := comp.ImageVulnerabilityCount(ctx, RawQuery{})
+			s.NoError(err)
+			s.Equal(expectedCount, count)
+
+			counter, err := comp.ImageVulnerabilityCounter(ctx, RawQuery{})
+			s.NoError(err)
+			s.Equal(expectedCount, counter.All(ctx).Total(ctx))
+			s.Equal(test.expectedCounterValues.fixable, counter.All(ctx).Fixable(ctx))
+			s.Equal(test.expectedCounterValues.critical, counter.Critical(ctx).Total(ctx))
+			s.Equal(test.expectedCounterValues.important, counter.Important(ctx).Total(ctx))
+			s.Equal(test.expectedCounterValues.moderate, counter.Moderate(ctx).Total(ctx))
+			s.Equal(test.expectedCounterValues.low, counter.Low(ctx).Total(ctx))
```

Comment:
By now, you probably are using in couple of test cases. Consider factoring it out into a func.
`func(ctx, t, expected, actual)`